x/epoching/keeper/msg_server_test.go

```diff
@@ -246,3 +247,80 @@ func FuzzMsgWrappedUpdateStakingParams(f *testing.F) {
 		require.Equal(t, newMinCommissionRate.String(), stakingParamsAfterChange.MinCommissionRate.String())
 	})
 }
+
+func TestExponentiallyEventsEndEpochQueuedMessages(t *testing.T) {
```

Comment:
To be clear, the number of events that were emitted in epoching module is exponential to the number of staking events, or linear? To my understanding is linear, i.e., for each staking event emitted in staking module, the epoching module will emit it again.

Reply:
it was exponential, in the test I wrote, for 7 delegations in epoch there was like ~300 events, for 12 there was like 140000events 😅